plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -388,6 +390,14 @@ public String quoted(RemoteTableName remoteTableName)
         return delegate.getTableScanRedirection(session, tableHandle);
     }
 
+    @Override
+    public OptionalLong delete(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        OptionalLong updateCount = delegate.delete(session, handle);
+        invalidateTableCaches(((JdbcTableHandle) handle).getSchemaTableName());
```

Comment:
why does this invalidation needed?

Reply:
I guess we might need to drop only the metadata/statistics if it exists. Other cache related to table meta need not be invalidated.

Reply:
We already do it this way for insert. So this is coherent.